### PR TITLE
docs: add security considerations

### PR DESCRIPTION
**Pull request by burgerdev**
Outlines a couple of DOs and DON'Ts when writing apps for Contrast. The idea is that, after reading the recommendations, an author of a Kubernetes app can tell what's safe to do in YAML and what needs scrutiny.

### CODE REVIEW

File: docs/docs/security-considerations.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+Contrast ensures application integrity and provides secure means of communication and bootstrapping (see [security benefits](basics/security-benefits.md)).
+However, care must be taken when interacting with the outside of Contrast's confidential environment.
+This page presents some tips for writing secure applications and outlines the trust boundaries app developers need to know.
+
+## General recommendations
+
+### Authentication
+
+The application receives credentials from the Contrast Coordinator during initialization.
+This allows to authenticate towards peers, and to verify credentials received from peers.
```
Comment (Freax13) [inline docs/docs/security-considerations.md:12]:
```suggestion
This allows to authenticate towards peers and to verify credentials received from peers.
```
Also in lines 13 and 38.

File: docs/docs/security-considerations.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+5. The containers' root filesystems are mounted in encrypted memory.
+
+### Limitations inherent to policy checking
+
+Workload policies serve as workload identities.
+From the perspective of the Contrast Coordinator, all workloads that authenticate with the same policy are equal.
+Thus, it's not possible to disambiguate, for example, pods spawned from a deployment, or to limit the amount of certificates issued per policy.
+
+Container image references from Kubernetes resource definitions are taken into account when generating the policy.
+A mutable reference may lead to policy failures or unverified image content, depending on the Contrast runtime.
```
Comment (Freax13) [inline docs/docs/security-considerations.md:41]: We should probably try to catch this and either log a warning or abort entirely.

Comment (m1ghtym0) [inline docs/docs/security-considerations.md:41]: I know that it's quite clear, but an example, like "docker.io/nginx:latest" or a link to some explaining resource could help.

**Review by katexochen — APPROVED** [review]

File: docs/docs/security-considerations.md (hunk 2 above)
Comment (burgerdev) [reply docs/docs/security-considerations.md:41]: Agreed, but I'd lean towards keeping the advice here for now and remove it once we agreed on a path forward (which is not obvious, considering other methods of integrity protection, like the tardev snapshotter or signed images).

Comment (Freax13) [reply docs/docs/security-considerations.md:41]: For sure, implementing that would be way out of the scope of this pr.

**Comment by burgerdev** [comment]
All feedback addressed.

**Review by m1ghtym0 — APPROVED** [review]
Thanks!
Did you envision this as a high-level section below the "workload deployment"?
Remember to add the sidebar entry before merging.

File: docs/docs/security-considerations.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+### Limitations inherent to policy checking
+
+Workload policies serve as workload identities.
+From the perspective of the Contrast Coordinator, all workloads that authenticate with the same policy are equal.
+Thus, it's not possible to disambiguate, for example, pods spawned from a deployment or to limit the amount of certificates issued per policy.
+
+Container image references from Kubernetes resource definitions are taken into account when generating the policy.
+A mutable reference may lead to policy failures or unverified image content, depending on the Contrast runtime.
+Reliability and security can only be ensured with a full image reference, including digest.
+The [`docker pull` documentation] explains pinned image references in detail.
```
[the remark this answers is not in the document]
Comment (burgerdev) [reply docs/docs/security-considerations.md:43]: That's a [shortcut reference link]. I like that it does not interrupt the sentence in the markdown source.

[shortcut reference link]: https://spec.commonmark.org/0.31.2/#shortcut-reference-link

**Comment by burgerdev** [comment]
> Thanks! Did you envision this as a high-level section below the "workload deployment"? Remember to add the sidebar entry before merging.

I reconsidered and moved it into *Architecture* for now - don't think this needs to be a top level item.

**Pull request merged**